From 07acca09912927325c157c1e8ca1e9dc46de9d2a Mon Sep 17 00:00:00 2001
From: Leszek Swirski <leszeks@chromium.org>
Date: Tue, 20 Jul 2021 14:04:10 +0200
Subject: [PATCH] [sparkplug] Simplify arch-guards to ENABLE_SPARKPLUG

This means we only have to maintain the arch list in one place.

Bug: v8:11420, v8:11421
Change-Id: I1618e2ab1396ad85aca60923111e5966e547e65a
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3041420
Auto-Submit: Leszek Swirski <leszeks@chromium.org>
Reviewed-by: Patrick Thier <pthier@chromium.org>
Commit-Queue: Leszek Swirski <leszeks@chromium.org>
Cr-Commit-Position: refs/heads/master@{#75813}
---
 src/baseline/baseline-assembler-inl.h | 9 ++++-----
 src/baseline/baseline-assembler.h     | 5 ++---
 src/baseline/baseline-compiler.cc     | 9 ++++-----
 src/baseline/baseline-compiler.h      | 7 +++----
 src/baseline/baseline.cc              | 5 ++---
 5 files changed, 15 insertions(+), 20 deletions(-)

diff --git a/src/baseline/baseline-assembler-inl.h b/src/baseline/baseline-assembler-inl.h
index 416fdbf98b..83c102176f 100644
--- a/src/baseline/baseline-assembler-inl.h
+++ b/src/baseline/baseline-assembler-inl.h
@@ -5,16 +5,15 @@
 #ifndef V8_BASELINE_BASELINE_ASSEMBLER_INL_H_
 #define V8_BASELINE_BASELINE_ASSEMBLER_INL_H_
 
+#include "src/baseline/baseline-assembler.h"
+
 // TODO(v8:11421): Remove #if once baseline compiler is ported to other
 // architectures.
-#if V8_TARGET_ARCH_IA32 || V8_TARGET_ARCH_X64 || V8_TARGET_ARCH_ARM64 ||     \
-    V8_TARGET_ARCH_ARM || V8_TARGET_ARCH_RISCV64 || V8_TARGET_ARCH_MIPS64 || \
-    V8_TARGET_ARCH_MIPS
+#if ENABLE_SPARKPLUG
 
 #include <type_traits>
 #include <unordered_map>
 
-#include "src/baseline/baseline-assembler.h"
 #include "src/codegen/interface-descriptors-inl.h"
 #include "src/interpreter/bytecode-register.h"
 #include "src/objects/feedback-cell.h"
@@ -142,6 +141,6 @@ SaveAccumulatorScope::~SaveAccumulatorScope() {
 }  // namespace internal
 }  // namespace v8
 
-#endif
+#endif  // ENABLE_SPARKPLUG
 
 #endif  // V8_BASELINE_BASELINE_ASSEMBLER_INL_H_
diff --git a/src/baseline/baseline-assembler.h b/src/baseline/baseline-assembler.h
index 8cafa8d94a..e1063ff2b2 100644
--- a/src/baseline/baseline-assembler.h
+++ b/src/baseline/baseline-assembler.h
@@ -7,9 +7,8 @@
 
 // TODO(v8:11421): Remove #if once baseline compiler is ported to other
 // architectures.
-#if V8_TARGET_ARCH_IA32 || V8_TARGET_ARCH_X64 || V8_TARGET_ARCH_ARM64 ||     \
-    V8_TARGET_ARCH_ARM || V8_TARGET_ARCH_RISCV64 || V8_TARGET_ARCH_MIPS64 || \
-    V8_TARGET_ARCH_MIPS
+#include "src/flags/flags.h"
+#if ENABLE_SPARKPLUG
 
 #include "src/codegen/macro-assembler.h"
 #include "src/objects/tagged-index.h"
diff --git a/src/baseline/baseline-compiler.cc b/src/baseline/baseline-compiler.cc
index e9874a321e..91276611b9 100644
--- a/src/baseline/baseline-compiler.cc
+++ b/src/baseline/baseline-compiler.cc
@@ -4,14 +4,13 @@
 
 // TODO(v8:11421): Remove #if once baseline compiler is ported to other
 // architectures.
-#include "src/base/bits.h"
-#if V8_TARGET_ARCH_IA32 || V8_TARGET_ARCH_X64 || V8_TARGET_ARCH_ARM64 ||     \
-    V8_TARGET_ARCH_ARM || V8_TARGET_ARCH_RISCV64 || V8_TARGET_ARCH_MIPS64 || \
-    V8_TARGET_ARCH_MIPS
+#include "src/flags/flags.h"
+#if ENABLE_SPARKPLUG
 
 #include <algorithm>
 #include <type_traits>
 
+#include "src/base/bits.h"
 #include "src/baseline/baseline-assembler-inl.h"
 #include "src/baseline/baseline-assembler.h"
 #include "src/baseline/baseline-compiler.h"
@@ -2249,4 +2248,4 @@ DEBUG_BREAK_BYTECODE_LIST(DEBUG_BREAK)
 }  // namespace internal
 }  // namespace v8
 
-#endif
+#endif  // ENABLE_SPARKPLUG
diff --git a/src/baseline/baseline-compiler.h b/src/baseline/baseline-compiler.h
index 68478804ea..bed0a9214e 100644
--- a/src/baseline/baseline-compiler.h
+++ b/src/baseline/baseline-compiler.h
@@ -7,9 +7,8 @@
 
 // TODO(v8:11421): Remove #if once baseline compiler is ported to other
 // architectures.
-#if V8_TARGET_ARCH_IA32 || V8_TARGET_ARCH_X64 || V8_TARGET_ARCH_ARM64 ||     \
-    V8_TARGET_ARCH_ARM || V8_TARGET_ARCH_RISCV64 || V8_TARGET_ARCH_MIPS64 || \
-    V8_TARGET_ARCH_MIPS
+#include "src/flags/flags.h"
+#if ENABLE_SPARKPLUG
 
 #include "src/base/logging.h"
 #include "src/base/threaded-list.h"
@@ -197,6 +196,6 @@ class BaselineCompiler {
 }  // namespace internal
 }  // namespace v8
 
-#endif
+#endif  // ENABLE_SPARKPLUG
 
 #endif  // V8_BASELINE_BASELINE_COMPILER_H_
diff --git a/src/baseline/baseline.cc b/src/baseline/baseline.cc
index be6e4ec3b3..cec0805aec 100644
--- a/src/baseline/baseline.cc
+++ b/src/baseline/baseline.cc
@@ -9,9 +9,8 @@
 
 // TODO(v8:11421): Remove #if once baseline compiler is ported to other
 // architectures.
-#if V8_TARGET_ARCH_IA32 || V8_TARGET_ARCH_X64 || V8_TARGET_ARCH_ARM64 ||     \
-    V8_TARGET_ARCH_ARM || V8_TARGET_ARCH_RISCV64 || V8_TARGET_ARCH_MIPS64 || \
-    V8_TARGET_ARCH_MIPS
+#include "src/flags/flags.h"
+#if ENABLE_SPARKPLUG
 
 #include "src/baseline/baseline-assembler-inl.h"
 #include "src/baseline/baseline-compiler.h"
-- 
2.35.1

